client/ayon_shotgrid/plugins/publish/integrate_shotgrid_publish.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pyblish.api
+
+from openpype.pipeline.publish import get_publish_repre_path
+
+
+class IntegrateShotgridPublish(pyblish.api.InstancePlugin):
+    """
+    Create published Files from representations and add it to version. If
+    representation is tagged as shotgrid review, it will add it in
+    path to movie for a movie file or path to frame for an image sequence.
+    """
+    order = pyblish.api.IntegratorOrder + 0.499
+    label = "Shotgrid Published Files"
+
+    def process(self, instance):
+        sg_session = instance.context.data.get("shotgridSession")
+        sg_version = instance.data.get("shotgridVersion")
+
+        if not sg_version:
+            return
+
+        for representation in instance.data.get("representations", []):
+
+            local_path = get_publish_repre_path(
+                instance, representation, False
+            )
+
+            if representation.get("tags", []):
+                continue
+
+            code = os.path.basename(local_path)
+
+            query_filters = [
+                ["project", "is", instance.data.get("shotgridProject")],
+                ["entity", "is", instance.data.get("shotgridEntity")],
+                ["version", "is", sg_version],
+                ["code", "is", code],
+            ]
+
+            if instance.data.get("shotgridTask"):
```

Comment:
```suggestion
            shotgrid_task =  instance.data.get("shotgridTask")
            if shotgrid_task:
```
And reuse it on the other places where you use the same

Reply:
same with other `instance.data.get("shotgridProject")` and `instance.data.get("shotgridEntity")`, create local variables